File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int j = word.indexOf(ch);

        if (j != -1) 
        {
            StringBuilder st = new StringBuilder(word.substring(0, j+1));
            st.reverse();
            return st.toString() + word.substring(j+1);
            
        }

        return word;

  
    }
}

File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.cpp
class Solution {
public:
    void reverseString(string& st, int left, int right) {
        while (left < right) {
            swap(st[left], st[right]);
            left++;
            right--;
        }
    }
    string reversePrefix(string word, char ch) {
        int indexFound = -1;
        for (int i = 0; i < word.size(); i++) {
            if (word[i] == ch) {
                indexFound = i;
                break;
            }
        }

        if (indexFound != -1)
            reverseString(word, 0, indexFound);

        return word;
    }
};